deployment/kube/prod/frontend.yaml

```diff
@@ -10,22 +10,28 @@ metadata:
     app: imageserver
 spec:
   replicas: 3
+  strategy:                                                                     
+    rollingUpdate:                                                              
+      maxSurge: 100%                                                            
+      maxUnavailable: 1                                                         
```

Comment:
Would it be reasonable to make this higher? In streambed we upgrade 25% of our nodes at a time (or that was the intention anyway), and if we lose an availability zone that's 33% of our nodes.

Reply:
if we have 3 pods running it will spin-up 3 new with the latest image and kill one by one the old one.

Reply:
yes its either a fixed number or a % so we can probably set it to 50%

Reply:
30% would be safer, unless you can guarantee that Kubernetes will wait for all the new nodes to become available before starting to remove nodes. (We wouldn't want to end up with 50% of the required number of nodes.)

Reply:
OK, the default sounds reasonable.